src/pagetable.h

```diff
@@ -18,17 +18,109 @@ See the file LICENSE for details.
 #define PAGE_FLAG_NOCLEAR     0
 #define PAGE_FLAG_CLEAR       8
 
+/**
+ * @brief   Create a pagetable
+ * @details Allocates a pagetable in memory. Each pagetable is one memory page
+ *          large, in this case, 4096 bytes. The pagetable needs to be manually
+ *          deleted using pagetable_delete.
+ *
+ * @return  A pointer to the newly created pagetable.
+ */
 struct pagetable *pagetable_create();
+
+/**
+ * @brief   Initialize a direct-mapped pagetable
+ * @details Initialize a given pagetable by direct mapping kernel space memory
+ *          up until 2GB, and direct mapping video memory for the video system.
+ *          TODO (SL): we should ensure video memory is correctly mapped into
+ *          kernel space even without vram in the future.
+ *
+ * @param   p A pointer to the pagetable to be initialized
+ */
 void pagetable_init(struct pagetable *p);
+
+/**
+ * @brief   Map a virtual address to a physical address in a page directory
+ * @details Given a virtual address, the pagetable maps its page to a given
+ *          physical frame  address. If PAGE_FLAG_ALLOC is passed in, paddr is
+ *          ignored, and a new physical frame is allocated to be mapped.
+ *
+ * @param   p       A pointer to the page directory to be modified
+ * @param   vaddr   Virtual address to be mapped
+ * @param   paddr   Physical address to be mapped. If PAGE_FLAG_ALLOC is passed
+                    in, paddr is ignored
+ * @param   flags   Flags of the new pages
+ * @return  1 if the mapping is successful, and 0 otherwise
+ */
 int pagetable_map(struct pagetable *p, unsigned vaddr, unsigned paddr,
                   int flags);
+
+/**
+ * @brief   Get the physical address of a virtual address in a page directory
+ * @details Given a virtual address, the pagetable gets its corresponding
+ *          physical address. The lookup is successful if the vaddr is mapped
+ *          to a paddr, and the page is currently in memory.
+ *
+ * @param   p       A pointer to the page directory to be looked up
+ * @param   vaddr   Virtual address to be looked up
+ * @param   paddr   A pointer to store the looked up physical address
+ * @return  1 if the lookup is successful, and 0 otherwise
```

Comment:
Can we add some discussion here of what a 0 indicates as a return value (here I believe it means that the location is not mapped)